### ISSUE TITLE
Warnings about missing initial state for anonymous components with a `:componentName`

### ISSUE THREAD

**Issue by MrEbbinghaus** [fulcrologic/fulcro#513]
Anonymous components have an empty initial state by default: (`{:initial-state (fn [& args] {})}`)

```clojure
(nc 
  [:car/id 
   {:car/owner [:user/id]]
  {:componentName ::Car})
```

Which leads to the warning: 

> Component user/Car does not INCLUDE initial state for anonymous at join key :car/owner ; however,  anonymous HAS initial state. This probably means your initial state graph is incomplete and props on anonymous will be nil.

**Comment by awkay** [comment]
What do you mean it has initial state?  The code you just provided creates a new component, which in fact, does not have initial state...nor is initial state even remotely what you'd want in this case, since such an on-the-fly component is never used in the initial state graph of the application (I mean, you can add initial-state to the options map if you're using this with `def` and composing it into that tree), but most uses of `nc` are well after the app initialization when initial state is used. Perhaps you're confusing your terminology and mean something else?

**Comment by awkay** [comment]
Oh sorry...you're complaining that the warning doesn't take into account that the given component doesn't need initial state? Sorry, not going to fix that. It has a constant ident, and you technically *could* use it with def and create the problem the warning is talking about.

**Comment by MrEbbinghaus** [comment]
I would say that it is very confusing when you get a warning about something that _might_ be a problem that you didn't even cause yourself, don't you think? 

Maybe the initial-state function of the anonymous component should include initial-state of its edges?
`{:car/owner (rc/initial-state <anonymous user component>)}` 

Or maybe components shouldn't have any implicit initial-state at all?

**Comment by awkay** [comment]
Hm. That is worth considering. I don't remember what I gave them a default initial state. At this point it would be a breaking change to remove it. So, I think we're stuck with the warning. I guess we could put a marker on `nc`-built components so the warning could skip them.

**Comment by MrEbbinghaus** [comment]
Would adding the missing edge to a child component in the initial-state be a breaking change?

**Comment by awkay** [comment]
Try 3.5.18-SNAPSHOT. I've turned off warning for nc  components altogether. That do it?

**Comment by awkay** [comment]
Fixed in 3.5.18
